server/app/controllers/budgetActions.js

```diff
@@ -25,8 +25,19 @@ const readBudgetById = async (req, res, next) => {
 
 const add = async (req, res, next) => {
   try {
-    const result = await tables.budget.create(req.body);
-    res.status(201).send(`Budget ajouté avec succès. ID : ${result.insertId}`);
+    const isCategory = await tables.category.readCategoryById({
+      id: req.body.category_id,
+      user_id: req.body.user_id,
+    });
+
+    if (isCategory) {
+      const result = await tables.budget.create(req.body);
+      res
+        .status(201)
+        .send(`Budget ajouté avec succès. ID : ${result.insertId}`);
+    } else {
+      res.status(401).send(`La catégorie n'existe pas.`);
```

Comment:
**question**: I don't think it's a 401, it would mean "Unauthorized", here it's mostly a 404 Not Found or a 400 Bad Request imo

Reply:
Even the error message points toward a 404 haha